microsetta_public_api/api/metadata.py

```diff
@@ -14,31 +14,33 @@ def category_values(category):
         return jsonify(text=text, error=404), 404
 
 
-def filter_sample_ids(taxonomy=None, alpha_metric=None, **kwargs):
-    repo = MetadataRepo()
-    query = _format_query(kwargs)
-    is_invalid = _validate_query(kwargs, repo)
-    if is_invalid:
-        return is_invalid
+def _filter_sample_ids(query, repo, alpha_metric, taxonomy):
     matching_ids = repo.sample_id_matches(query)
-
     matching_ids, error_code, error_response = _filter_matching_ids(
         matching_ids, TaxonomyRepo, 'resources', taxonomy, 'resource',
     )
-
     matching_ids, error_code, error_response = _filter_matching_ids(
         matching_ids, AlphaRepo, 'available_metrics', alpha_metric,
         'metric', error_response=error_response, error_code=error_code,
     )
-
     if error_response:
         return error_response, error_code
-
     return jsonify(sample_ids=matching_ids), 200
 
 
+def filter_sample_ids(taxonomy=None, alpha_metric=None, **kwargs):
+    repo = MetadataRepo()
+    query = _format_query(kwargs)
+    is_invalid = _validate_query(kwargs, repo)
+    if is_invalid:
+        return is_invalid
+    return _filter_sample_ids(query, repo, alpha_metric, taxonomy)
+
+
 def filter_sample_ids_query_builder(body, taxonomy=None, alpha_metric=None):
-    raise NotImplementedError()
+    query = body
```

Comment:
given the API spec, does this actually come in over the signature? I assumed this needed to be obtained off of `request`. But regardless, isn't the `.json` what's desired here?

Reply:
also I think the use of `query` in `_filter_sample_ids` is quire a bit different than what the example in the spec shows right?

Reply:
`query` here is fed directly to `MetadataRepo.sample_id_matches(query)`, which I explicitly designed to accept data that looks like the `QueryBuilder` outputs.

The metadata endpoint we designed, here: https://github.com/biocore/microsetta-public-api/blob/bf4d17307c501a55089c1d0f79425b2964b7c879/microsetta_public_api/api/microsetta_public_api.yml#L77-L96

directly takes the `QueryBuilder`-formatted query as the body of the request, which is exactly what the spec shows in this example:
https://github.com/biocore/microsetta-public-api/blob/bf4d17307c501a55089c1d0f79425b2964b7c879/microsetta_public_api/api/microsetta_public_api.yml#L626-L667

Reply:
You can also consult the API tests for a reference on acceptable inputs for this operation:
https://github.com/biocore/microsetta-public-api/blob/bf4d17307c501a55089c1d0f79425b2964b7c879/microsetta_public_api/api/tests/test_api.py#L246-L356

Reply:
I see now... I think flask is just rather permissive in how you interact with body. And ya, mislooked at use with `_filter_sample_ids`